Chap_Introduction.tex

```diff
@@ -143,71 +74,112 @@ \section{PMIx Architecture Overview}
-
-Recognizing the burden this places on SMS vendors, the PMIx community has included interfaces by
-which the host can request support from local SMS elements. Once the SMS has transferred the request to
-an appropriate location, a PMIx server interface can be used to pass the request between SMS subsystems.
-For example, a request for network traffic statistics can utilize the
-PMIx networking abstractions to retrieve the information from the FM. This reduces the portability and
-interoperability issues between the individual subsystems by transferring the burden of defining the
-interoperable interfaces from the SMS subsystems to the PMIx community, which continues
-to work with those providers to develop the necessary support.
-
-Tools, whether standalone or embedded in job scripts, are an exception to the communication rule and can connect to
-any PMIx server providing they are given adequate rendezvous information. The PMIx conceptual model views the
-collection of PMIx servers as a cloud-like conglomerate --- i.e., orchestration and information requests can be
-given to any server regardless of location. However, tools frequently execute on locations that may not house an
-operating PMIx server --- e.g., a users notebook computer. Thus, tools need the ability to remotely connect to
-the PMIx server ``cloud''.
+Fig.~\ref{fig:roles} shows a typical \ac{PMIx} implementation in which the application is 
+built against a \ac{PMIx} client library that contains the client-side \acp{API},
+attribute definitions, and communication support for interacting with the local \ac{PMIx} server. 
+\ac{PMIx} clients are processes which are started through the \ac{PMIx} infrastructure, 
+either by the PMIx implementation directly or through an \ac{SMS} component, and have registered 
+as clients.  A \ac{PMIx} client 
+is created in such a way that the \ac{PMIx} client library
+will be have sufficient information available to
+authenticate with the \ac{PMIx} server.  
+The \ac{PMIx} server will have sufficient knowledge about the  
+process which it created, either directly or through other \ac{SMS}, to authenticate the 
+process and provide information the process requests such as its identity and the 
+identity of its peers.  
+
+As clients invoke \ac{PMIx} \acp{API}, is possible that some client requests can 
+be handled at the client level.  Other requests might require communication with the 
+local \ac{PMIx} server, which subsequently might request services from the host \ac{SMS} 
+(represented here by an \ac{RM} daemon).  
+
+The interaction between the \ac{PMIx} server and \ac{SMS} might be 
+achieved using callback functions registered during server initialization.  
+The host \ac{SMS} could indicate its lack of support for any operation by simply providing a \textit{NULL} for the associated callback function, or could create a function entry that returns \textit{not supported} when called.  These interactions between a \ac{PMIx} implementation 
```

Comment:
Are these callback functions for the server-side interface?

Reply:
see above